images/airflow/2.9.2/python/mwaa/entrypoint.py

```diff
@@ -602,13 +602,18 @@ def run_airflow_command(cmd: str, environ: Dict[str, str]):
             subprocesses = _create_airflow_scheduler_subprocesses(environ, conditions)
             # Schedulers, triggers, and DAG processors are all essential processes and
             # if any fails, we want to exit the container and let it restart.
-            run_subprocesses(subprocesses, essential_subprocesses=subprocesses)
+            run_subprocesses(
```

Comment:
In the `run_subprocesses` implementation, the docs mention that
```
:param subprocesses: A list of Subprocess objects to run in parallel.
    :param essential_subprocesses: A sub-list of the processes that that must continue
      running, otherwise all sub-processes will be terminated.
```

But then, later this behavior got altered when this line was introduced within run_subprocesses
```
all_processes = subprocesses + essential_subprocesses
```
essential_subprocesses was supposed to be a sub list of subprocesses. 

I think a cleaner approach would be to make create_airflow_subprocess method accept an additional parameter named `essential` which can be used by the run_subprocesses method to check if a failed condition should cause the container to exit or not instead of passing two separate array of subprocesses.